### PR TITLE
Just call Arrange to set the Frame

### LINKED ISSUE #8146: FlyoutHeader hangs ios application on startup
### Description

Adding a Layout inside of the Shell.FlyoutHeader is causing the application to hang/crash on ios simulator and device

[Repo](https://github.com/BaldeGH/MauiFlyoutBug)

image

### Steps to Reproduce

1. Create a new maui program
2. Add LayoutControl to AppShell.xaml's Shell.FlyoutHeader
3. Build & Deploy

### Version with bug

6.0.400 (current)

### Last version that worked well

6.0.312

### Affected platforms

iOS

### Affected platform versions

iOS 15.4

### Did you find any workaround?

Remove the VerticalStackLayout and only allow a single item in the flyout

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by PureWeen**
### Description of Change

The following [PR ](https://github.com/dotnet/maui/pull/7297) added a `SetNeedsLayout` call that will propagate up to the FlyoutHeader container during LayoutSubviews. This causes an infinite loop so this stops the call to `SetNeedsLayout` if the call comes from `LayoutSubviews`

I doubt this is the best fix but it's the fix that gets it back to not crashing. The best fix will come once this issue is fixed https://github.com/dotnet/maui/issues/8568

### Issues Fixed

Fixes #8146

### Testing info

[We have tests for this but we're still figuring out why shell tests don't run on CI .](https://github.com/dotnet/maui/issues/8568)

### CODE REVIEW

File: src/Controls/src/Core/Compatibility/Handlers/Shell/iOS/UIContainerView.cs [hunk 1]
```diff
@@ -90,9 +90,27 @@ public override void WillMoveToSuperview(UIView newSuper)
...
+				base.SetNeedsLayout();
+		}
+
 		public override void LayoutSubviews()
 		{
-			_view.LayoutToSize(Width ?? Frame.Width, Height ?? MeasuredHeight);
+			try
+			{
+				_currentlyInLayoutSubviews = true;
+				_view.LayoutToSize(Width ?? Frame.Width, Height ?? MeasuredHeight);
```
Comment (hartez) [inline src/Controls/src/Core/Compatibility/Handlers/Shell/iOS/UIContainerView.cs:108]: Just a guess, but I think the problem is that LayoutToSize is doing something weird where it manually sets the Frame for the platform view and _then_ calls Arrange on the IView. I think if you remove that explicit Frame settting and just call Arrange, it won't get stuck in a loop.

Comment (PureWeen) [reply src/Controls/src/Core/Compatibility/Handlers/Shell/iOS/UIContainerView.cs]: Alright, if I just rely on arrange things seem to all behave a bit better 👍

File: src/Controls/src/Core/Compatibility/Handlers/Shell/iOS/ShellFlyoutLayoutManager.cs [hunk 2]
```diff
@@ -257,12 +256,11 @@ public void LayoutParallax()
 				{
 					topMargin = (float)UIApplication.SharedApplication.GetSafeAreaInsetsForWindow().Top;
 				}
+				else if (OperatingSystem.IsIOSVersionAtLeast(11))
```
Comment (jsuarezruiz) [inline src/Controls/src/Core/Compatibility/Handlers/Shell/iOS/ShellFlyoutLayoutManager.cs:259]: Why only on iOS 11 or higher?

Comment (PureWeen) [reply src/Controls/src/Core/Compatibility/Handlers/Shell/iOS/ShellFlyoutLayoutManager.cs]: Good catch!! That was left over from when I was just checking safe area directly here, opposed to the header margin, which is only part of iOS11

**Review by hartez — APPROVED** [review]

**Pull request merged**